app/main.py

```diff
@@ -1,3 +1,72 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | Distance) -> Distance:
+        if isinstance(other, (int, float)):
+            new_distance = Distance(other)
+            return Distance(self.km + new_distance.km)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: int | Distance) -> Distance:
+        if isinstance(other, (int, float)):
+            new_distance = Distance(other)
+            self.km = self + new_distance
+            return self
+        self.km = self + other
+        return self
+
+    def __mul__(self, other: int | float | Distance) -> Distance:
+        if isinstance(other, (int, float, Distance)):
+            new_distance = Distance(other)
+            return Distance(self.km * new_distance.km)
+        return Distance(self.km * other.km)
```

Comment:
duplicates code. try to name a new instance of Distance as "other". Hence, you will need only one return statement